### PR TITLE
feat: add ControlBar plugin

### PR DESCRIPTION
**Pull request by WoodNeck**
## Details
This adds a new plugin - "ControlBar"

### CODE REVIEW

File: src/plugin/ControlBar/AnimationProgressBar.ts [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+    };
+
+    const root = document.createElement(BROWSER.EL_DIV);
+    root.classList.add(className.PROGRESS_ROOT);
+    root.draggable = false;
+
+    const track = document.createElement(BROWSER.EL_DIV);
+    track.classList.add(className.PROGRESS_TRACK);
+
+    const knob = document.createElement(BROWSER.EL_DIV);
```
Comment (daybrush) [inline src/plugin/ControlBar/AnimationProgressBar.ts:145]: Maybe the knob is a thumb?

Sliders and video players seem to use the word "thumb" more often.

File: src/plugin/ControlBar/AnimationProgressBar.ts [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+    const knob = this._knobEl;
+
+    const animator = view3D.animator;
+    const activeAnimationIdx = animator.activeAnimationIndex;
+    const activeAnimationClip = animator.activeAnimation;
+    const activeAnimationAction = animator.actions[activeAnimationIdx];
+
+    if (!activeAnimationClip || !activeAnimationAction) return;
+
+    const progress = ((x - rootBbox.x) / rootBbox.width);
```
Comment (daybrush) [inline src/plugin/ControlBar/AnimationProgressBar.ts:308]: If there is a scroll left, would the value be different?

Comment (WoodNeck) [reply src/plugin/ControlBar/AnimationProgressBar.ts:303]: Good point, I checked & fixed the wrong behavior of the animation progress bar when there's a scroll bar that exists outside the canvas area.

**Review by malangfox — APPROVED** [review]
LGTM

**Pull request merged**